giftless/transfer/basic_streaming.py

```diff
@@ -144,9 +149,16 @@ def download(self, organization: str, repo: str, oid: str, size: int,
 
         else:
             download_url = ObjectsView.get_storage_url('get', organization, repo, oid)
+            preauth_url = self._preauth_url(download_url, organization, repo, actions={'read'}, oid=oid)
+
+            params = {'filename': extra.get('filename') if extra else None}
+            qs = urlencode(params)
+            sep = '&' if '?' in preauth_url else '?'
+            preauth_url_with_params = f'{preauth_url}{sep}{qs}'
```

Comment:
This is reuse of code from [here](https://github.com/datopian/giftless/blob/59e6b0a5373558947382d69e891780f41dd88225/giftless/transfer/__init__.py#L58).  It's only a few lines, but could be refactored.  Just wasn't sure where would be best to put it?

Reply:
If it's more than 2 lines, it's worth deduplicating :) 
I think `util.py` is a good place, and you can also add some doctest to it if you wish!